pilot/pkg/model/push_context.go

```diff
@@ -1403,9 +1403,13 @@ func (ps *PushContext) updateContext(
 
 // Caches list of services in the registry, and creates a map
 // of hostname to service
-func (ps *PushContext) initServiceRegistry(env *Environment) {
+func (ps *PushContext) initServiceRegistry(env *Environment, configsUpdate sets.Set[ConfigKey]) {
 	// Sort the services in order of creation.
 	allServices := SortServicesByCreationTime(env.Services())
+	if features.EnableExternalNameAlias {
+		resolveServiceAliases(allServices, configsUpdate)
```

Comment:
If we enable this flag, from performance perspective this is costy. So why cannot this be done in service conversion

Reply:
Because it needs to join across all services of all types in all registries. I am not sure its so costly, its only a O(services) traversal in majority of cases

Reply:
however it is always doing such work. I will review other part though